### PR TITLE
Order Edit: Remove feature flag

### LINKED ISSUE #4786: Order Editing: Release Feature
After all tasks are completed we can release the feature to the public

- [x] Do a quick run-through of the features to make sure we didn't miss anything and there are no critical bugs. If available, we can use Sheri's testing guide. Non-critical bugs don't need to block the release. They can be fixed later.
- [x] Remove feature flag
- [x] Update release notes
- [x] Notify testers

### PR DESCRIPTION
**Pull request by Ecarrion**
closes #4786 

# Why

The feature is now stable and useful for merchants. 

We will be releasing this as it is for `7.7` and will continue tackling outstanding improvements in the next sprint.

# How

- Remove feature flag and the conditionals that it generated.
- Update Release notes

Update release notes:
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewModels/Order Details/OrderDetailsViewModel.swift [hunk 1]
```diff
@@ -264,7 +264,7 @@ extension OrderDetailsViewModel {
             viewController.show(productListVC, sender: nil)
         case .billingDetail:
             ServiceLocator.analytics.track(.orderDetailShowBillingTapped)
-            let billingInformationViewController = BillingInformationViewController(order: order, editingEnabled: dataSource.orderEditingEnabled)
+            let billingInformationViewController = BillingInformationViewController(order: order)
```
Comment (ealeksandrov) [inline WooCommerce/Classes/ViewModels/Order Details/OrderDetailsViewModel.swift:267]:
```suggestion
            let billingInformationViewController = BillingInformationViewController(order: order, editingEnabled: true)
```

File: WooCommerce/Classes/ViewRelated/Orders/Order Details/Customer Section/Billing Information/BillingInformationViewController.swift [hunk 2]
```diff
@@ -18,15 +18,10 @@ final class BillingInformationViewController: UIViewController {
...
-    let editingEnabled: Bool
-
     /// Designated Initializer
     ///
-    init(order: Order, editingEnabled: Bool = false) {
+    init(order: Order) {
         self.order = order
-        self.editingEnabled = editingEnabled
         super.init(nibName: type(of: self).nibName, bundle: nil)
     }
```
Comment (ealeksandrov) [inline WooCommerce/Classes/ViewRelated/Orders/Order Details/Customer Section/Billing Information/BillingInformationViewController.swift]: This is also enabling editing on `ReviewOrderViewController`, where we don't want edit indicators.

File: WooCommerce/Classes/ViewModels/Order Details/OrderDetailsViewModel.swift (hunk 1 above)
Comment (ealeksandrov) [inline WooCommerce/Classes/ViewModels/Order Details/OrderDetailsViewModel.swift]:
```suggestion
            let billingInformationViewController = BillingInformationViewController(order: order, editingEnabled: true)
```

**Review by ealeksandrov — COMMENTED** [review]
I think we don't need edit actions in `ReviewOrderViewController`. Added highlights in code!

File: WooCommerce/Classes/ViewRelated/Orders/Order Details/Customer Section/Billing Information/BillingInformationViewController.swift (hunk 2 above)
Comment (Ecarrion) [reply WooCommerce/Classes/ViewRelated/Orders/Order Details/Customer Section/Billing Information/BillingInformationViewController.swift]: Oh nice catch! I totally missed it!

**Comment by Ecarrion** [comment]
Thanks for noticing the regression @ealeksandrov.
♻️

**Review by ealeksandrov — APPROVED** [review]

**Pull request merged**